src/main.rs

```diff
@@ -7,7 +19,182 @@ use ic_cdk::api::management_canister::http_request::{HttpHeader, HttpResponse, T
 use ic_cdk::{query, update};
 use ic_nervous_system_common::{serve_logs, serve_logs_v2, serve_metrics};
 
-use eth_rpc::*;
+use evm_rpc::*;
+use serde::de::DeserializeOwned;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct CanisterTransport;
+
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+impl RpcTransport for CanisterTransport {
+    fn get_subnet_size() -> u32 {
+        METADATA.with(|m| m.borrow().get().nodes_in_subnet)
+    }
+
+    fn resolve_api(provider: RpcNodeProvider) -> Result<RpcApi, ProviderError> {
+        // fn resolve_by_hostname(hostname: &str) -> Result<RpcApi, ProviderError> {
```

Comment:
more dead code?

Reply:
This is a placeholder for giving the Candid-RPC logic (shared with ckETH) a way to access API keys once they're available, which is turning into quite a large refactor. I'm planning to complete this in a separate PR.

Reply:
How about adding a comment and a linked Jira ticket for the sake of clarity?